src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,490 @@
+            temp_file_path.push(format!("nix-installer-tmp.{}", rng.gen::<u32>()));
+        }
+        let mut temp_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            // If the file is created, ensure that it has harmless
+            // permissions regardless of whether the mode will be
+            // changed later (if we ever create setuid executables,
+            // they should only become setuid once they are owned by
+            // the appropriate user)
+            .mode(0o600)
+            .open(&temp_file_path)
+            .await
+            .map_err(|e| {
+                ActionError::Open(temp_file_path.clone(), e)
+            })?;
+
+        // FIXME(@cole-h): for now we replace the entire file, but in the future we could potentially "replace" the contents
+        let mut new_config = String::new();
+        if let Some(existing_nix_config) = &nix_configs.existing_nix_config {
+            for (name, value) in existing_nix_config.settings() {
+                if nix_configs.merged_nix_config.settings().get(name).is_some() {
+                    continue;
+                }
+
+                new_config.push_str(name);
+                new_config.push_str(" = ");
+                new_config.push_str(value);
+                new_config.push('\n');
+            }
+
+            new_config.push('\n');
+        }
+
+        new_config.push_str(&format!(
+            "# Generated by https://github.com/DeterminateSystems/nix-installer, version {version}.\n",
+            version = env!("CARGO_PKG_VERSION"),
+        ));
+
+        for (name, value) in nix_configs.merged_nix_config.settings() {
+            new_config.push_str(name);
+            new_config.push_str(" = ");
+            new_config.push_str(value);
+            new_config.push('\n');
+        }
+
+        temp_file
+            .write_all(new_config.as_bytes())
+            .await
+            .map_err(|e| ActionError::Write(temp_file_path.clone(), e))?;
+        tokio::fs::set_permissions(&temp_file_path, PermissionsExt::from_mode(NIX_CONF_MODE))
+            .await
+            .map_err(|e| ActionError::SetPermissions(NIX_CONF_MODE, path.to_owned(), e))?;
+        tokio::fs::rename(&temp_file_path, &path)
+            .await
+            .map_err(|e| ActionError::Rename(temp_file_path.to_owned(), path.to_owned(), e))?;
+
+        Ok(())
+    }
+
+    fn revert_description(&self) -> Vec<ActionDescription> {
+        let Self {
+            path,
+            nix_configs: _,
+        } = &self;
+
+        vec![ActionDescription::new(
+            format!("Delete file `{}`", path.display()),
+            vec![format!("Delete file `{}`", path.display())],
+        )]
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn revert(&mut self) -> Result<(), ActionError> {
+        let Self {
+            path,
+            nix_configs: _,
+        } = self;
+
+        remove_file(&path)
```

Comment:
Noting for myself later that this is correct as due to #126 we do in fact just want to remove this.